components/app.js

```diff
@@ -47,11 +44,30 @@ class App {
     this.gradeTable.onEditClick(this.gradeForm.setFormValue);
 
   }
+  findIndexwithId(id) {
+    for (var i = 0; i < this.cachedGrades.length; i++) {
+      if (id === this.cachedGrades[i].id) {
+        return i
+      }
+    }
+  }
+  success() {
+    this.gradeTable.updateGrades(this.cachedGrades);
+    var total = 0;
+    for (var i = 0; i < this.cachedGrades.length; i++) {
+      total += this.cachedGrades[i].grade
+    }
+    var average = total / this.cachedGrades.length;
+
+    this.pagerHeader.updateAverage(parseInt(average));
+  }
```

Comment:
Not something you need to change, but when you get to React, a method which only refreshes the content on the page is called `render`.